### PR TITLE
Add config scope integration to ActiveSolutionBoundTracker

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Part of #5240

### CODE REVIEW

File: src/Integration.UnitTests/MefServices/ConfigScopeUpdaterTests.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
```
Comment (pavel-mikula-sonarsource) [inline src/Integration.UnitTests/MefServices/ConfigScopeUpdaterTests.cs:24]: You desperately need global usings :)
https://github.com/SonarSource/sonar-dotnet/blob/72e14ee23ad494c38c1f6d80c3155991ee21f6eb/analyzers/tests/SonarAnalyzer.Test/SonarAnalyzer.Test.csproj#L76C3-L81C46

What can be added globally, add it to your targets for tests. What can be added only locally to individual CSPRoj, add it only there (maybe Moq is not everywhere, I don't know)

Comment (pavel-mikula-sonarsource) [reply src/Integration.UnitTests/MefServices/ConfigScopeUpdaterTests.cs:24]: ..not in this PR

File: src/Integration/MefServices/ConfigScopeUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    private async Task HandleConfigScopeUpdateInternalAsync(string solutionName, string connectionId, string projectKey)
+    {
+        if (solutionName is null)
+        {
+            await activeConfigScopeTracker.RemoveCurrentConfigScopeAsync();
+            return;
+        }
+
+        await activeConfigScopeTracker.SetCurrentConfigScopeAsync(solutionName, connectionId, projectKey);
```
Comment (pavel-mikula-sonarsource) [inline src/Integration/MefServices/ConfigScopeUpdater.cs:76]: I have a strong preference for `if ... else ...` in this case

File: src/SLCore.UnitTests/Common/Helpers/ConnectionIdHelperTests.cs [hunk 3]
```diff
@@ -20,12 +20,26 @@
...
 namespace SonarLint.VisualStudio.SLCore.UnitTests.Common.Helpers;
 
 [TestClass]
 public class ConnectionIdHelperTests
 {
+    [TestMethod]
+    public void MefCtor_CheckIsExported()
+    {
+        MefTestHelpers.CheckTypeCanBeImported<ConnectionIdHelper, IConnectionIdHelper>();
+    }
```
Comment (pavel-mikula-sonarsource) [inline src/SLCore.UnitTests/Common/Helpers/ConnectionIdHelperTests.cs:34]: What's your opinion on arrow methods and their general adoption in the codebase?

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.UnitTests/Common/Helpers/ConnectionIdHelperTests.cs:34]: I am pro-arrow methods in general, but in the case of tests I am against it

File: src/Integration.UnitTests/MefServices/ConfigScopeUpdaterTests.cs (hunk 1 above)
Comment (ugras-ergun-sonarsource) [reply src/Integration.UnitTests/MefServices/ConfigScopeUpdaterTests.cs:24]: I think we have them in newer libraries....

**Review by ugras-ergun-sonarsource — APPROVED** [review]

**Pull request merged**